From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sat, 9 Mar 2024 17:39:20 +0000
Subject: [PATCH] =?UTF-8?q?Extended=20asm=20with=20C=20operands=20doesn?=
 =?UTF-8?q?=E2=80=99t=20work=20at=20top=20level?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

fix the following test in the meson.build

stderr:
testfile.c:6:34: error: expected ')' before '::' token
    6 |__asm__ (""::)
      |        ~  ^~
      |           )
testfile.c:6:37: error: expected ';' at end of input
    6 |__asm__ (""::)
      |              ^
      |              ;
-----------
Checking if "compiler supports gcc-style inline assembly" compiles: NO

refer:
- https://gcc.gnu.org/bugzilla/show_bug.cgi?id=41045

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>
Signed-off-by: Jean-Marc Valin <jmvalin@jmvalin.ca>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 5623260962fc..1b16b08f662a 100644
--- a/meson.build
+++ b/meson.build
@@ -227,7 +227,7 @@ if not opt_asm.disabled()
                       #error GCC before 3.4 has critical bugs compiling inline assembly
                       #endif
                       #endif
-                      __asm__ (""::)''',
+                      int main(int argc, char ** argv) { __asm__ (""::); }''',
                    name : 'compiler supports gcc-style inline assembly')
 
       opus_conf.set('OPUS_ARM_INLINE_ASM', 1)
